src/test/scala/DiceBotTest.scala

```diff
@@ -0,0 +1,14 @@
+import org.scalatest.FunSuite
+
+class DiceBotTest extends FunSuite {
+  Seq(
+    "invalid string",
+    null,
+    "").foreach(x => {
+    test(s"process with invalid input '$x' returns none.") {
+      val expected = None
+      val actual = DiceBot.process("invalid string")
+      assert(expected == actual)
```

Comment:
Hm. The idea is from [the parameterized test](http://xunitpatterns.com/Parameterized%20Test.html). Usually tests that you have encountered are [Tabular Test](http://xunitpatterns.com/Parameterized%20Test.html#Tabular%20Test) when you need tests but only test parameters are different from each others.

The following quotation sharply describes what is the different between the parameterized test and the tabular test.

> Several early reviewers wrote to me about a variation they use regularly: the Tabular Test. The essence of this is the same as doing a Parameterized Test except that the entire table of values is in a single Test Method. Unfortunately, this makes the test an Eager Test (see Assertion Roulette on page X) because it verifies many test conditions. This isn't a problem when all the tests are passing but it does lead to a lack of Defect Localization (see Goals of Test Automation) when one of the rows fails.